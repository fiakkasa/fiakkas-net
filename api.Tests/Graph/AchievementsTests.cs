using HotChocolate.Execution;

namespace api.Tests.Graph;

public class AchievementsTests(GraphFixture fixture) : IClassFixture<GraphFixture>
{
    [Fact]
    public async Task Achievements_Should_Return_Data()
    {
        var executor = await fixture.GetRequestExecutor();

        var result = await executor.ExecuteAsync(
            """
            {
              achievements {
                totalCount
                nodes {
                  content
                  createdAt
                  id
                  internalId
                  updatedAt
                  version
                  years
                  yearsSummary
                }
              }
            }
            """);

        var fn = result.ExpectQueryResult;
        fn.Should().NotThrow();
        fn().Errors.Should().BeNullOrEmpty();
        result.ToJson().MatchSnapshot();
    }

    [Fact]
    public async Task AchievementById_Should_Return_Data()
    {
        var executor = await fixture.GetRequestExecutor();

        var result = await executor.ExecuteAsync(
            """
            {
              achievementById(id: "QWNoaWV2ZW1lbnQKZ2Q0NjA1YjBjNThiYzQ5YWNiY2ZkMTBhMjRhMjAzYWRk") {
                content
                createdAt
                id
                internalId
                updatedAt
                version
                years
                yearsSummary
              }
            }
            """);

        var fn = result.ExpectQueryResult;
        fn.Should().NotThrow();
        fn().Errors.Should().BeNullOrEmpty();
        result.ToJson().MatchSnapshot();
    }
}
