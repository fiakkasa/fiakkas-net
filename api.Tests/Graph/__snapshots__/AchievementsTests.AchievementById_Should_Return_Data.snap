{
  "data": {
    "achievementById": {
      "content": "Content",
      "createdAt": "2024-01-01T00:00:00.000Z",
      "id": "QWNoaWV2ZW1lbnQ6DFtg1LxYrEm8/RCiSiA63Q==",
      "internalId": "d4605b0c-58bc-49ac-bcfd-10a24a203add",
      "updatedAt": null,
      "version": 1,
      "years": [
        2024
      ],
      "yearsSummary": "2024"
    }
  }
}
